indico/modules/rb_new/client/js/components/RoomBasicDetails.jsx

```diff
@@ -80,11 +93,32 @@ export default class RoomBasicDetails extends React.PureComponent {
         return (
             <Grid columns={2}>
                 <Grid.Column textAlign="center" styleName="photo-column">
-                    <div styleName="image-container">
-                        <SpriteImage pos={room.spritePosition}
+                    <Modal open={openImageModal}
+                           onClose={() => this.setState({openImageModal: false})}>
+                        <Modal.Content image styleName="image-dimmer">
+                            <Dimmer.Dimmable>
+                                <Dimmer active={!imageLoaded} inverted>
+                                    <Loader />
+                                </Dimmer>
+                                <Image wrapped
+                                       onLoad={() => this.setState({imageLoaded: true})}
+                                       src={fetchRoomImageURL({room_id: room.id})} />
+                            </Dimmer.Dimmable>
+                        </Modal.Content>
+                    </Modal>
+                    <div styleName={isClickable ? 'image-container clickable-image' : 'image-container'}>
+                        <SpriteImage styleName="clickable-image"
+                                     pos={room.spritePosition}
                                      height="100%"
                                      width="100%"
-                                     fillVertical />
+                                     fillVertical
+                                     onClick={() => {
```

Comment:
This could also be split into a `renderImage` method. That way `render` doesn't become so huge and if you define the `<SpriteImage>` at the beginning, you can return it wrapped in the div straight away if the image isn't clickable. No need to always have the modal stuff around.